### PR TITLE
STM - Static stm implementation

### PR DESCRIPTION
**Pull request by DH254**
Implementation do allow both a static and dynamic pod run using transitions as parameter

### CODE REVIEW

**Review by TomLonergan03 — APPROVED** [review]

File: lib/state_machine/state_machine.cpp [hunk 1]
```diff
@@ -4,7 +4,9 @@
 
 namespace hyped::state_machine {
 
-StateMachine::StateMachine() : current_state_{State::kIdle}
+StateMachine::StateMachine(const TransitionTable &transition_table)
+    : current_state_{State::kIdle},
```
Comment (DylanCavers) [inline lib/state_machine/state_machine.cpp:8]: This was my mistake a while ago, quick fix here
```suggestion
    : current_state_(State::kIdle),
```

File: lib/state_machine/state_machine.cpp [hunk 2]
```diff
@@ -4,7 +4,9 @@
...
+      transition_to_state_{transition_table}
```
Comment (DylanCavers) [inline lib/state_machine/state_machine.cpp:9]:
```suggestion
      transition_to_state_(transition_table)
```

**Review by DylanCavers — COMMENTED** [review]
will just make these changes then looks good

**Review by DylanCavers — APPROVED** [review]
lovely stuff

**Pull request merged**